### ISSUE TITLE
Content gets sometimes duplicated

### ISSUE THREAD

**Issue by olafgrabienski** [backdrop-contrib/expanding_formatter#2]
Content in the expanded area gets duplicated in certain situations, such as content with headings or lists. (I'm not sure about the exact conditions.)

To reproduce, configure the Body field in a Page default display with the "Trimmed (expandable)" format, create a page with some text and a list (see screenshots below), and hit "Expand".

Result: Parts of the content get duplicated.

## Screenshots

**(a) Collapsed:**
<img src="https://github.com/user-attachments/assets/8949f8fd-c75c-4b39-b452-f74b564209f1" width=220>

**(b) Expanded:**
<img src="https://github.com/user-attachments/assets/2dbe5e61-133f-493e-bf43-3521c817331c" width=220>

**Comment by laryn** [comment]
@olafgrabienski I haven't been able to replicate this, but I've made a few additional tweaks separately (not released yet but in the latest two commits). Let me know if those seem to make any difference or if you have specific steps to reproduce on a fresh site.

**Comment by olafgrabienski** [comment]
@laryn Strange, I _was_ able to reproduce the issue shown in the screenshot on a fresh Demo site (without the last two commits).

If you want to have a look, I've just installed the site, and you can see the issue on the About page:
https://latest-eekgtm8p39mgkogdsrvhzrnclakjgfsn.tugboatqa.com/about

**Comment by laryn** [comment]
Yes, I see. It does seem related to HTML within the page in some way. Maybe a simplification would be just to show/hide the summary and hide/show the entire default content instead of trying to cut and splice those back together.

**Comment by olafgrabienski** [comment]
Simplification sounds good. For me it's not urgent, by the way, I'm not using the module (but still interested).

**Comment by oadaeh** [comment]
@olafgrabienski and @laryn, when you tried to duplicate this, had you set "Trim length" to 0 in the Manage Displays tab of the content type?
Here: Administration > Structure > Content types > $content_type > Manage displays > Default / Manage display
Edit: I should have added, try setting "Trim length" to both 0 and to 1 to see if either change the behavior.

**Comment by olafgrabienski** [comment]
@oadaeh If I recall correctly, I didn't change the default value of `200` at that time. (I guess it means a length of 200 characters. Are you saying, this setting is maybe switching between `1` and `0` instead, or is supposed to switch that way?)

Anyways, when I change the Trim length to `1`, the body text disappears. There is only an ellipsis and an Expand link, and when I click on "Expand", nothing gets duplicated.

When I change the Trim length to `0`, the "Expand link" is at the end of the body text, and it seems the whole text gets duplicated.

**Comment by oadaeh** [comment]
@olafgrabienski what you are seeing with lengths of 0 and 1 is what I am seeing, and I wondered if that was what you were doing previously.

I had not tested with a different length previously, but I just did try it (with 100, 200, 250, and 500), and I am not seeing duplicated text with any value greater than 0 like I am with a length of 0.

The code doesn't treat any number other than 0 differently, so I didn't expect any non-0 value to behave differently than any other non-0 value, which is why I only tested with 1 (that and because I was testing to see if using a value of 1 would get me the same results I expected with 0, and it did).

I tested with both 1.x-1.1.0 and 1.x-1.x and got the same results. I have a fix for the problem I am seeing, but not the one you are seeing, and so I was also checking to see if I should open a new issue or add my patch here.

**Comment by olafgrabienski** [comment]
@oadaeh It's interesting that you don't see duplicated text at all with lenghts like 200. My impression is however that the issue happens rather with slightly more complex content, e.g. text which starts with a heading or with a list.
